default.nix

```
@@ -20,10 +23,23 @@ let
   #       };
   #     "other-package": { ... };
   #   }
-  snapshotFromPackageLockJson = packageLockJson:
+  snapshotFromPackageLockJson = packageLockJson: pname: version:
     let
       packageLock = builtins.fromJSON (builtins.readFile packageLockJson);
 
+      # Load custom name and version of the program in case it was specified and
+      # not specified by the package-lock.json
+      topPackageName =
+          if builtins.hasAttr "name" packageLock then packageLock.name
+          else if ! isNull pname then pname
+          else fallbackPackageName;
```

I beleive this should work as expected:
```suggestion
      ifNotNull = a: b: if ! isNull a then a else b; # there must be something like this already in lib somewhere 
      topPackageName = packageLock.name or (ifNotNull pname fallbackPackageName);
```